ui/src/treeDataProvider/AzureTreeDataProvider.ts

```diff
@@ -39,14 +40,16 @@ export class AzureTreeDataProvider implements TreeDataProvider<IAzureNode>, Disp
 
     private _disposables: Disposable[] = [];
 
-    constructor(resourceProvider: IChildProvider, loadMoreCommandId: string, rootTreeItems?: IAzureParentTreeItem[]) {
+    constructor(resourceProvider: IChildProvider, loadMoreCommandId: string, rootTreeItems?: IAzureParentTreeItem[], testAccount?: TestAzureAccount) {
         this._resourceProvider = resourceProvider;
         this._loadMoreCommandId = loadMoreCommandId;
         this._customRootNodes = rootTreeItems ? rootTreeItems.map((treeItem: IAzureParentTreeItem) => new RootNode(this, treeItem, this._onNodeCreateEmitter)) : [];
 
         // Rather than expose 'AzureAccount' types in the index.ts contract, simply get it inside of this npm package
         const azureAccountExtension: Extension<AzureAccount> | undefined = extensions.getExtension<AzureAccount>('ms-vscode.azure-account');
-        if (!azureAccountExtension) {
+        if (testAccount) {
```

Comment:
The only issue I'm having with using a debug env variable is trying to create a TestAzureAccount in the constructor since I can't call async functions within it.  I'm not entirely sure how to remedy this without passing the account directly to the AzureTreeDataProvider.

Reply:
There can actually be advantages to having the TestAzureAccount around - it can provide additional testing functionality, such as being told to log in and out, etc.  

Like Eric, I'd love to see the client PR for this, too, though.